### PR TITLE
One vs rest on MNIST

### PR DESCRIPTION
**Pull request by aditiagarwal-02**
Training a simple MLP for the one-vs-rest classification problem on MNIST dataset

### CODE REVIEW

**Comment by patel-zeel** [comment]
Please remove `requirements.txt` from this PR.

File: sandbox/one_vs_rest-aditi.ipynb [hunk 1]
```diff
@@ -0,0 +1,355 @@
+{
```
Comment (patel-zeel) [inline sandbox/one_vs_rest-aditi.ipynb:1]: <p>Training with MLP is a good first step. Add a CNN to this notebook to compare the performance. Also add a RF classifier.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/patel-zeel/albk/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (patel-zeel) [inline sandbox/one_vs_rest-aditi.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    # Some images from train dataset
</pre><p>Why one image is missing here?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/patel-zeel/albk/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (patel-zeel) [inline sandbox/one_vs_rest-aditi.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    # Hyperparameters
</pre><p>Model definition and training can be kept in the same cell if we don't want to continue training by rerunning a cell. That becomes ambiguous and thus defining and training the model in the same cell makes more sense.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/patel-zeel/albk/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (patel-zeel) [inline sandbox/one_vs_rest-aditi.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #7.    model = MLP(784, 500, 2)
</pre><p>When we instantiate the model, it also initializes the weights. Train it for multiple random initialization and report the accuracy on each of the seed as well as the average accuracy and standard deviation in accuracy.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/patel-zeel/albk/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (patel-zeel) [inline sandbox/one_vs_rest-aditi.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.    for epoch in range(num_epochs):
</pre><p>We can also use tqdm to see the progress. It also has an option to add custom text (e.g. loss values). We should always plot the loss v/s iterations after any training.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/patel-zeel/albk/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (patel-zeel) [inline sandbox/one_vs_rest-aditi.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #13.        print("Accuracy: {} %".format(100 * correct / total))
</pre><p>Can you plot the calibration curve and also compute expected calibration error here? Do this for all baselines and at the end we should plot them all in a single plot for a comparison and questions like, is CNN better calibrated than MLP?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/patel-zeel/albk/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

**Comment by patel-zeel** [comment]
Once the above-suggested changes are completed, please run the brick kiln classification task on the same models (except for the input channel adjustment needed for the RGB image in the CNN model). Then, please observe the model's behavior on test metrics with different random initializations.

File: sandbox/one_vs_rest-aditi.ipynb (hunk 1 above)
Comment (nipunbatra) [reply sandbox/one_vs_rest-aditi.ipynb:1]: <p>I agree. Plot loss v/s epochs</p>

Comment (nipunbatra) [inline sandbox/one_vs_rest-aditi.ipynb:1]: <p>Make the code generic so that we can quickly test for other configurations.</p><p> </p><p>Like, </p><p><code>positive_class = 3</code>  <span class="ql-cursor">﻿</span></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/patel-zeel/albk/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (nipunbatra) [inline sandbox/one_vs_rest-aditi.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    # Some images from train dataset
</pre><p>Better do <code>fig, ax = plt.subplots( ...)</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/patel-zeel/albk/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (nipunbatra) [reply sandbox/one_vs_rest-aditi.ipynb:1]: <p>Beyond this, we want to analyse the results and not end at the accuracy numbers.</p><p> </p><p>Plot (and analyse) the images:</p><ul><li>we predict wrong with high probability</li><li>we predict wrong with low probability</li><li>we predict right with high probability</li></ul><p> </p><p>Plot the histogram of predicted probabilities for all samples and then individually for each class. </p>
